substrate/frame/staking/src/pallet/mod.rs

```diff
@@ -986,6 +967,150 @@ pub mod pallet {
 		}
 	}
 
+	impl<T: Config> Pallet<T> {
+		/// Get the ideal number of active validators.
+		pub fn validator_count() -> u32 {
+			ValidatorCount::<T>::get()
+		}
+
+		/// Get the minimum number of staking participants before emergency conditions are imposed.
+		pub fn minimum_validator_count() -> u32 {
+			MinimumValidatorCount::<T>::get()
+		}
+
+		/// Get the validators that may never be slashed or forcibly kicked out.
+		pub fn invulnerables() -> Vec<T::AccountId> {
+			Invulnerables::<T>::get()
+		}
+
+		/// Get the preferences of a given validator.
+		pub fn validators<EncodedAccountId>(account_id: EncodedAccountId) -> ValidatorPrefs
```

Comment:
@kianenigma cargo expand shows that FRAME macros actually expand to these (seemingly over-) generic function signatures, is this just to allow args to be passed as a ref or value, or is there another reason? I guess we also need them here to not be a breaking change, or can we make assumptions about usage here?